tests/integration/workflows/test_workflows_tasks_refextract.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2014-2018 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+from __future__ import absolute_import, division, print_function
+
+from inspirehep.modules.workflows.tasks.refextract import match_reference
+
+from factories.db.invenio_records import TestRecordMetadata
+
+
+def test_match_reference_for_jcap_and_jhep_config():
+    """Test reference matcher for the JCAP and JHEP configuration"""
+
+    cited_record_json = {
+        '$schema': 'http://localhost:5000/schemas/records/hep.json',
+        '_collections': ['Literature'],
+        'control_number': 1,
+        'document_type': ['article'],
+        'publication_info': [
+            {
+                'artid': '045',
+                'journal_title': 'JHEP',
+                'journal_volume': '06',
+                'page_start': '045',
+                'year': 2007
+            }
+        ],
+        'titles': [
+            {
+                'title': 'The Strongly-Interacting Light Higgs'
+            }
+        ],
+    }
+
+    TestRecordMetadata.create_from_kwargs(
+        json=cited_record_json, index_name='records-hep')
+
+    reference = {
+        'reference': {
+            'publication_info': {
+                'artid': '045',
+                'journal_title': 'JHEP',
+                'journal_volume': '06',
+                'page_start': '045',
+                'year': 2007
+            }
+        }
+    }
+
+    reference = match_reference(reference)
+
+    assert reference['record']['$ref'] == 'http://localhost:5000/api/literature/1'
+    assert isinstance(reference['reference']['publication_info']['year'], int)
```

Comment:
I meant actually call the `validate` function, like here https://github.com/inspirehep/inspire-next/blob/16504109805e9b93b034ef5cf4f50683ea55b579/tests/unit/workflows/test_workflows_tasks_refextract.py#L130